### ISSUE TITLE
False positive warning on if keyword for unused_qualifications

### ISSUE THREAD

**Issue by djc** [rust-lang/rust#149873]
### Code

See here: 

https://github.com/hickory-dns/hickory-dns/blob/ac67742d64717cf61299871554590beb7c0c296c/crates/proto/src/multicast/mdns_stream.rs#L456

(After enabling `unused_qualifications = "warn"` via `Cargo.toml`.)

### Current output

```Shell
warning: unnecessary qualification
   --> crates/proto/src/multicast/mdns_stream.rs:456:9
    |
456 |         if let Err(error) = result {
    |         ^^
    |
    = note: requested on the command line with `-W unused-qualifications`

warning: `hickory-proto` (lib test) generated 1 warning (run `cargo fix --lib -p hickory-proto --tests` to apply 1 suggestion)
```

### Desired output

```Shell
No warning.
```

### Rationale and extra context

<img width="943" height="500" alt="Image" src="https://github.com/user-attachments/assets/1c62bc38-b4fd-46c4-8b22-9cf128575cf7" />

Clearly the lint should not fire on language keywords.

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: aarch64-apple-darwin
release: 1.92.0
LLVM version: 21.1.3
```

**Comment by mu001999** [comment]
> the branch from this PR (after removing the suppression)

@djc It has been merged, could you provide a specific commit which can reproduce this directly?

**Comment by mu001999** [comment]
MCVE:

```
pub use tokio::runtime;

#[tokio::test]
pub async fn test_next_random_socket() {
    if true {}
}
```

And we will get:
```
warning: unnecessary qualification
 --> src/lib.rs:5:5
  |
5 |     if true {}
  |     ^^
  |
  = note: requested on the command line with `-W unused-qualifications`
```
